modules/openapi-generator/src/main/resources/python-nextgen/model_generic.mustache

```diff
@@ -0,0 +1,258 @@
+
+class {{classname}}({{#parent}}{{{.}}}{{/parent}}{{^parent}}BaseModel{{/parent}}):
+    """NOTE: This class is auto generated by OpenAPI Generator.
+    Ref: https://openapi-generator.tech
+
+    Do not edit the class manually.
+    """
+{{#vars}}
+    {{name}}: {{{vendorExtensions.x-py-typing}}}
+{{/vars}}
+{{#isAdditionalPropertiesTrue}}
+    additional_properties: Dict[str, Any] = {}
+{{/isAdditionalPropertiesTrue}}
+    __properties = [{{#allVars}}"{{baseName}}"{{^-last}}, {{/-last}}{{/allVars}}]
+{{#vars}}
+    {{#vendorExtensions.x-regex}}
+
+    @validator('{{{name}}}')
+    def {{{name}}}_validate_regular_expression(cls, v):
+        if not re.match(r"{{{.}}}", v{{#vendorExtensions.x-modifiers}} ,re.{{{.}}}{{/vendorExtensions.x-modifiers}}):
+            raise ValueError(r"must validate the regular expression {{{pattern}}}")
+        return v
+    {{/vendorExtensions.x-regex}}
+    {{#isEnum}}
+
+    @validator('{{{name}}}')
+    def {{{name}}}_validate_enum(cls, v):
+        {{^required}}
+        if v is None:
+            return v
+
+        {{/required}}
+        if v not in ({{#allowableValues}}{{#enumVars}}{{{value}}}{{^-last}}, {{/-last}}{{/enumVars}}{{/allowableValues}}):
+            raise ValueError("must validate the enum values ({{#allowableValues}}{{#enumVars}}{{{value}}}{{^-last}}, {{/-last}}{{/enumVars}}{{/allowableValues}})")
+        return v
+    {{/isEnum}}
+{{/vars}}
+
+    class Config:
+        allow_population_by_field_name = True
+        validate_assignment = True
+
+{{#hasChildren}}
+{{#discriminator}}
+    # JSON field name that stores the object type
+    __discriminator_property_name = '{{discriminator.propertyBaseName}}'
+
+    {{#mappedModels}}
+    {{#-first}}
+    # discriminator mappings
+    __discriminator_value_class_map = {
+    {{/-first}}
+        '{{{mappingName}}}': '{{{modelName}}}'{{^-last}},{{/-last}}
+    {{#-last}}
+    }
+
+    @classmethod
+    def get_discriminator_value(cls, obj: dict) -> str:
+        """Returns the discriminator value (object type) of the data"""
+        discriminator_value = obj[cls.__discriminator_property_name]
+        if discriminator_value:
+            return cls.__discriminator_value_class_map.get(discriminator_value)
+        else:
+            return None
+    {{/-last}}
+    {{/mappedModels}}
+
+{{/discriminator}}
+{{/hasChildren}}
+    def to_str(self) -> str:
+        """Returns the string representation of the model using alias"""
+        return pprint.pformat(self.dict(by_alias=True))
+
+    def to_json(self) -> str:
+        """Returns the JSON representation of the model using alias"""
+        return json.dumps(self.to_dict())
+
+    @classmethod
+    def from_json(cls, json_str: str) -> {{^hasChildren}}{{{classname}}}{{/hasChildren}}{{#hasChildren}}{{#discriminator}}Union({{#children}}{{{classname}}}{{^-last}}, {{/-last}}{{/children}}){{/discriminator}}{{^discriminator}}{{{classname}}}{{/discriminator}}{{/hasChildren}}:
```

Comment:
what if model properties need to be named
- from_json
- to_json
- from_dict

How about adding a trailing suffix `_` to allow that use case?

Reply:
We can whitelist those if needed when users report such use (edge) cases. I whitelisted some when testing some specs shared by our users previously)

(FYI. pydantic will skip validations for field names starting with `_`)